pyresample/spherical.py

```diff
@@ -125,12 +196,18 @@ def norm(self):
 
     def normalize(self):
         """Normalize the vector."""
-        self.cart /= np.sqrt(np.einsum('...i, ...i', self.cart, self.cart))
-
-        return self
+        # Note: if self.cart == [0,0,0], norm=0 and cart becomes [nan, nan, nan]
+        norm = self.norm()
+        cart = self.cart / norm[..., np.newaxis]  # enable vectorization
+        # mask_norm_0 = norm == 0
+        # cart[mask_norm_0, ...] = 0 # This remove nan propagation
+        return CCoordinate(cart)
```

Comment:
The original method normalizes in place, this is changing the legacy behaviour.

Reply:
I added an inplace argument

Reply:
@mraspaud do we need backwards compatibility with this? I'm fine with having a flag for now, but I think in general inplace operations feel like bad practice and should be avoided, especially by default.